astropy/coordinates/tests/test_name_resolve.py

```diff
@@ -136,6 +140,38 @@ def test_names():
     np.testing.assert_almost_equal(icrs.dec.degree, icrs_true.dec.degree, 1)
 
 
+@pytest.mark.remote_data
+def test_name_resolve_cache(tmpdir):
+    from astropy.utils.data import _get_download_cache_locs
+    import shelve
+
+    temp_cache_dir = str(tmpdir.mkdir('cache'))
+    with paths.set_temp_cache(temp_cache_dir, delete=True):
+        download_dir, urlmapfn = _get_download_cache_locs()
+
+        with shelve.open(urlmapfn) as url2hash:
+            assert len(url2hash) == 0
+
+        icrs1 = get_icrs_coordinates("castor", cache=True)
+
+        # This is a weak test: we just check to see that a url is added to the
```

Comment:
Why not have a stronger test with [astropy.utils.data.get_cached_urls](https://docs.astropy.org/en/latest/api/astropy.utils.data.get_cached_urls.html#astropy.utils.data.get_cached_urls)?

Reply:
Because I didn't know about that :) - I'll add a new check